rasa/core/migrator.py

```diff
@@ -0,0 +1,214 @@
+        self.event_broker = event_broker
+        self.requested_conversation_ids = requested_conversation_ids
+        self.minimum_timestamp = minimum_timestamp
+        self.maximum_timestamp = maximum_timestamp
+
+    def publish_events(self) -> int:
+        """Publish events in a tracker store using an event broker.
+
+        Exits if the publishing of events is interrupted due to an error. In that case,
+        the CLI command to continue the export where it was interrupted is printed.
+
+        """
+        events = self._fetch_events_within_time_range()
+
+        cli_utils.print_info(
+            f"Selected {len(events)} events for publishing. Ready to go 🚀"
+        )
+
+        published_events = 0
+        current_timestamp = None
+
+        for event in tqdm(events, "events"):
+            # noinspection PyBroadException
+            try:
+                body = {"sender_id": event["sender_id"]}
+                body.update(event)
+                self.event_broker.publish(body)
+                published_events += 1
+                current_timestamp = event["timestamp"]
+            except Exception as e:
+                logger.exception(e)
+                raise PublishingError(current_timestamp)
+
+        self.event_broker.close()
+
+        return published_events
+
+    def get_conversation_ids_to_process(self) -> Set[Text]:
+        """Get conversation IDs that are good for processing.
+
+        Finds the intersection of events that are contained in the tracker store with
+        those events requested as a command-line argument.
+
+        Prints an error and if no conversation IDs are found in the tracker, or if no
+        overlap is found between those contained in the tracker and those requested
+        by the user.
+
+        Returns:
+            Conversation IDs that are both requested and contained in the tracker
+            store. If no conversation IDs are requested, all conversation IDs in the
+            tracker store are returned.
+
+        """
+        conversation_ids_in_tracker_store = set(self.tracker_store.keys())
+
+        if not conversation_ids_in_tracker_store:
+            raise NoConversationsInTrackerStore(
+                f"Could not find any conversations in connected tracker store. "
+                f"Please validate your `endpoints.yml` and make sure the defined "
+                f"tracker store exists. Exiting."
+            )
+
+        if not self.requested_conversation_ids:
+            return conversation_ids_in_tracker_store
+
+        missing_ids_in_tracker_store = (
+            set(self.requested_conversation_ids) - conversation_ids_in_tracker_store
+        )
+
+        if missing_ids_in_tracker_store:
+            cli_utils.print_warning(
+                f"Could not find the following requested "
+                f"conversation IDs in connected tracker store: "
+                f"{', '.join(sorted(missing_ids_in_tracker_store))}"
```

Comment:
why sort them? Who know if they were actually passed in sorted?

Reply:
so it's reproducible - I think it would be confusing if you retry the export command, it keeps failing, and each time IDs come back in a different order